File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/pincode/PincodeUseCase.java
package nl.acidcats.tumblrlikes.core.usecases.pincode;


import rx.Observable;

/**
 * Created on 28/09/2018.
 */
public interface PincodeUseCase {
    Observable<Boolean> checkPincode(String pincode);

    Observable<Boolean> storePincode(String pincode);

    Observable<Boolean> isAppPincodeProtected();
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/usecases/pincode/PincodeUseCase.kt
package nl.acidcats.tumblrlikes.core.usecases.pincode


import rx.Observable

/**
 * Created on 28/09/2018.
 */
interface PincodeUseCase {

    fun isAppPincodeProtected(): Observable<Boolean>

    fun checkPincode(pincode: String): Observable<Boolean>

    fun storePincode(pincode: String): Observable<Boolean>
}
